plugins/anx-udev/fu-anx-udev-firmware.c

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2021 Xiaotian Cui <xtcui@analogixsemi.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-anx-udev-common.h"
+#include "fu-anx-udev-firmware.h"
+
+
+struct _FuAnxUdevFirmware {
+	FuFirmwareClass  parent_instance;
+};
+
+G_DEFINE_TYPE (FuAnxUdevFirmware, fu_anx_udev_firmware, FU_TYPE_FIRMWARE)
+
+
+static gboolean
+fu_anx_udev_firmware_parse (FuFirmware *firmware,
+                           GBytes *fw,
+                           guint64 addr_start,
+                           guint64 addr_end,
+                           FwupdInstallFlags flags,
+                           GError **error)
+{
+	guint8* bin_buf = g_malloc0 (MAX_FILE_SIZE);
+	/*guint32 payload_len;*/
+	g_autofree gchar *version = NULL;
+	g_autoptr(FuFirmwareImage) img_hdr = fu_firmware_image_new (NULL);
+	g_autoptr(FuFirmwareImage) img_payload = fu_firmware_image_new (NULL);

```

Comment:
We're not using FuFirmwareImage in master anymore, I think you want FuFirmware instead